nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SplitText.java

```diff
@@ -150,548 +145,320 @@
-                            System.arraycopy(headerBytes, 0, headerBytesWithoutTrailingNewLines, 0, headerBytes.length - headerNewLineByteCount);
+                        long stop = System.nanoTime();
+                        if (getLogger().isDebugEnabled()) {
+                            getLogger().debug("Computed splits in " + (stop - start) + " milliseconds.");
                         }
-                    } else {
-                        headerBytesWithoutTrailingNewLines = null;
-                        headerNewLineBytes = null;
                     }
-
-                    while (true) {
-                        if (headerInfoLineCount > 0) {
-                            // if we have header lines, create a new FlowFile, copy the header lines to that file,
-                            // and then start copying lines
-                            final AtomicInteger linesCopied = new AtomicInteger(0);
-                            final AtomicLong bytesCopied = new AtomicLong(0L);
-                            FlowFile splitFile = session.create(flowFile);
-                            try {
-                                splitFile = session.write(splitFile, new OutputStreamCallback() {
-                                    @Override
-                                    public void process(final OutputStream rawOut) throws IOException {
-                                        try (final BufferedOutputStream out = new BufferedOutputStream(rawOut);
-                                                final ByteCountingOutputStream countingOut = new ByteCountingOutputStream(out)) {
-                                            countingOut.write(headerBytesWithoutTrailingNewLines);
-                                            //readLines has an offset of countingOut.getBytesWritten() to allow for header bytes written already
-                                            linesCopied.set(readLines(in, maxLineCount, maxFragmentSize - countingOut.getBytesWritten(), countingOut,
-                                                    includeLineDelimiter, headerNewLineBytes));
-                                            bytesCopied.set(countingOut.getBytesWritten());
-                                        }
-                                    }
-                                });
-                                splitFile = session.putAttribute(splitFile, SPLIT_LINE_COUNT, String.valueOf(linesCopied.get()));
-                                splitFile = session.putAttribute(splitFile, FRAGMENT_SIZE, String.valueOf(bytesCopied.get()));
-                                logger.debug("Created Split File {} with {} lines, {} bytes", new Object[]{splitFile, linesCopied.get(), bytesCopied.get()});
-                            } finally {
-                                if (linesCopied.get() > 0) {
-                                    splits.add(splitFile);
+                }
+            });
+            if (!error.get()) {
```

Comment:
We should fail-fast here - if error.get() then transfer to failure and return, rather than indenting the next 50 lines or so within a new block